linkerd/app/outbound/src/tcp/concrete.rs

```diff
@@ -1,13 +1,25 @@
 use super::{Concrete, Endpoint};
-use crate::{endpoint, logical::ConcreteError, stack_labels, Outbound};
+use crate::{endpoint, stack_labels, Outbound};
 use linkerd_app_core::{
     drain, io,
-    proxy::{api_resolve::Metadata, core::Resolve, tcp},
+    proxy::{
+        api_resolve::{ConcreteAddr, Metadata},
+        core::Resolve,
+        tcp,
+    },
     svc, Error,
 };
 use std::time;
 use tracing::info_span;
 
+#[derive(Debug, thiserror::Error)]
+#[error("concrete service {addr}: {source}")]
+pub struct ConcreteError {
+    addr: ConcreteAddr,
+    #[source]
+    source: Error,
+}
```

Comment:
what's the motivation for having separate versions of `ConcreteError`, `LogicalError`, and `EndpointError` for the HTTP and TCP stacks when they are now identical?

Reply:
> All error types are defined in the module in which they are used, even when they exactly match error types elsewhere. As we move forward with the upcoming changes, we want each stack to be as independent as possible.

As I mentioned when we did a read-through of https://github.com/linkerd/linkerd2-proxy/tree/ver/policy-client-sketch, the current strategy (to support modularity, refactoring, etc) is to focus on decoupling over reuse. We'll be doing the same type of decoupling with target types.

If they really are common, we should _also_ split out the common parts of the stack impl so that a single module can hold it all.

Basically: we should drive towards making each stack module as self-contained as possible.